plugins/hls-explicit-imports-plugin/src/Ide/Plugin/ExplicitImports.hs

```diff
@@ -129,13 +145,19 @@ runImportCommand _ _ _ rd = do
 -- the provider should produce one code lens associated to the import statement:
 -- > Refine imports to import Control.Monad.IO.Class (liftIO)
 lensProvider :: Recorder (WithPriority Log) -> PluginMethodHandler IdeState 'Method_TextDocumentCodeLens
-lensProvider _  state _ CodeLensParams {_textDocument = TextDocumentIdentifier {_uri}} = do
-    nfp <- getNormalizedFilePathE _uri
-    (ImportActionsResult{forLens}, pm) <- runActionE "ImportActions" state $ useWithStaleE ImportActions nfp
-    let lens = [ generateLens _uri newRange int
-                | (range, int) <- forLens
-                , Just newRange <- [toCurrentRange pm range]]
-    pure $ InL lens
+lensProvider _ state _ CodeLensParams {_textDocument = TextDocumentIdentifier {_uri}} = do
+    -- Code lens are not provided when the client supports inlay hints,
+    -- otherwise it will be provided as a fallback
+    isIHSupported <- liftIO $ isInlayHintsSupported state
+    if isIHSupported
+    then do pure $ InL []
```

Comment:
Comment about why this is the right thing to return here

Reply:
I'm trying to figure out what's the difference between the two, and from this comment (https://github.com/microsoft/language-server-protocol/issues/1200#issuecomment-779094756), it seems like `[]` is no different from `null`.

But we do need to uniformly choose a way to indicate “no information”.  Do you think Right `[]` is better?

Reply:
Yes, that seems reasonable to me.